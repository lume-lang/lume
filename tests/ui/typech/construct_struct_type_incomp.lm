struct A {
    pub foo: Boolean;
}

fn foo() {
    let a = A { foo: 0 };
}
