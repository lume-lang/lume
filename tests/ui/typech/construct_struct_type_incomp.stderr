× error[LM4001]: mismatched types
   ╭─[construct_struct_type_incomp.lm:6:22]
 5 │ fn foo() {
 6 │     let a = A { foo: 0 };
   │                      ^ expected type Boolean, but found type Int32...
 7 │ }
   ∶
 1 │ struct A {
 2 │     foo: Boolean;
   │          ^^^^^^^ ...because of type defined here
 3 │ }
   ╰──
   help: expected type Boolean
            found type Int32